pkg/timeout/handler_test.go

```diff
@@ -111,7 +111,7 @@ func TestTaskRunCheckTimeouts(t *testing.T) {
 	}
 
 	th.SetCallbackFunc(f)
-	th.CheckTimeouts(context.Background(), testNs, c.Kube, c.Pipeline)
+	go th.CheckTimeouts(context.Background(), testNs, c.Kube, c.Pipeline)
```

Comment:
The test already use a polling approach below via `wait.PollImmediate` so Its okay not to block here. However, if leaving this here non blocking is a concern, im happy to block until the call is done to preserve old behavior.